### PR TITLE
HTTP to TChannel forwarding demo

### PR DESCRIPTION
**Pull request by kriskowal**
These changes add three example services: an HTTP caller, an HTTP to TChannel proxy, and a TChannel service. The proxy is encoding agnostic and just translates the transport enveloping. The demo can be run locally or inside a Docker composition.

### CODE REVIEW

**Comment by breerly** [comment]
Can you merge dev in then move this code down into `internal/examples`?

**Comment by kriskowal** [comment]
@breerly moved to internal

**Comment by breerly** [comment]
@kriskowal is this under test coverage somehow? Like the other examples, I worry this will get stale.

File: internal/examples/proxy/http2tchannel/proxy/main.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	defer inbound.Stop()
+
+	err = outbound.Start(transport.NoDeps)
+	if err != nil {
+		log.Fatalf("failed to start outbound %v", err)
+	}
+	defer outbound.Stop()
+
+	log.Printf("server started\n")
+	select {}
```
Comment (breerly) [inline internal/examples/proxy/http2tchannel/proxy/main.go]: What features does one lose when they don't use the dispatcher?

File: internal/examples/proxy/http2tchannel/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+This is an example that illustrates an HTTP to TChannel transport translation
+proxy for a single service in a Docker composition.
+The "caller" services sends HTTP requests, the "proxy" service receives HTTP
+requests and forwards them over TChannel to the "service".
+
+To run this example, make the Linux binaries for each of the services
+(`*/main.go`) and then bring up the services in Docker.
+
+```
+make
```
Comment (breerly) [inline internal/examples/proxy/http2tchannel/README.md]: You should update the readme to show that they need to have Go install and have this repo checked out in the correct place in their `$GOPATH`.

File: internal/examples/proxy/http2tchannel/caller/service/service_procedure.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
+// Code generated by thriftrw v1.0.0
```
Comment (breerly) [inline internal/examples/proxy/http2tchannel/caller/service/service_procedure.go]: <3<3<3

File: internal/examples/proxy/http2tchannel/caller/service/versioncheck.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package service
+
+import "go.uber.org/thriftrw/version"
+
+func init() {
+	version.CheckCompatWithGeneratedCodeAt("1.0.0", "go.uber.org/yarpc/examples/proxy/http2tchannel/caller/service")
```
Comment (breerly) [inline internal/examples/proxy/http2tchannel/caller/service/versioncheck.go]: *swoon* <3

File: internal/examples/proxy/http2tchannel/proxy/main.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	}
+
+	flag.Parse()
+	inbound := yhttp.NewInbound(flag.Arg(0))
+	outbound := ytchannel.NewOutbound(channel, ytchannel.HostPort(flag.Arg(1)))
+
+	detail := transport.ServiceDetail{
+		Name: "proxy",
+		Registry: &registry{
+			outbound: transport.NewUnaryRelay(outbound),
```
Comment (breerly) [inline internal/examples/proxy/http2tchannel/proxy/main.go]: Does this registry need to be updated to be:

```
&registry{
    outbound{
        unary: transport.NewUnaryRelay(outbound),
    }
}
```

I just want to make sure we're leaving room for the other RPC Types.

Comment (breerly) [inline internal/examples/proxy/http2tchannel/proxy/main.go]: You know what - I think I was confused by the term `Registry`...

**Pull request closed without merging**